hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/AbstractUpgradeDowngrade.java

```diff
@@ -132,6 +134,8 @@ protected void run(HoodieTableVersion toVersion, String instantTime) throws IOEx
   }
 
   private void createUpdatedFile(Properties props) throws IOException {
+    DefaultHoodieConfig.setDefaultOnCondition(props, !props.containsKey(HoodieIndexConfig.INDEX_TYPE_PROP),
```

Comment:
just confirming: ideally there would be an index type set already in props, in most cases. so thats what gets written. not the default, in that case

Reply:
yes, but have case such that :
before the upgrade, there is no index type in hoodie properties, can persist the index type as DEFAULT_INDEX_TYPE in hoodie properties after upgrade

Reply:
but should we persist the configured index type as opposed to the default? they can be different right

Reply:
got it